ui/ui_v1/scripts/process_query.js

```diff
@@ -0,0 +1,236 @@
+   */
+  JsonObj.isAsc = isAsc;
+  
+  /* 
+   * set limit
+   * JsonObj.k is an Integer storing K 
+   * (if user doesn't specify values of k -5000000 default)
+   */
+  JsonObj.k = k;
+   
+  /*
+   * set slice
+   * JsonObj.slices is a 1d array of objects
+   * each object has 3 keys - sliceCol, sliceOp, sliceVal
+   * sliceCol is a string, sliceOp is a string, sliceVal is 1d array of strings
+   */
+  if(slices.length > 0)
+    JsonObj.slices = slices;
+   
+  /*
+   * set dateRange
+   * JsonObj.dateRange is an object with 3 keys - dateCol, dateStart, dateEnd
+   * dateCol is a string, dateStart and dateEnd are dates (yyyy-mm-dd) 
+   */
+  if(dateRange.dateCol !== '' && dateRange.dateStart !== '' && dateRange.dateEnd !== '') {
+    dateRange.dateStart = new Date(dateRange.dateStart);
+    dateRange.dateEnd = new Date(dateRange.dateEnd);
+    JsonObj.dateRange=dateRange;
+    //formatDate(dateRange.dateCol);
+  }
+    
+  /*
+   * set timeGranularity
+   * JsonObj.timeGranularity is a string containing the selected value
+   */
+  JsonObj.timeGranularity = timeGranularity;
+  
+  //printing the JsonObj
+  Logger.log(JsonObj);
+
+  //printing JsonObj in json format
+  Logger.log(JSON.stringify(JsonObj));
+  
+  //calling GCP function
+  var outputTable = callGcpFunc();
+  
+  //converting date from json into javascript date object
+  outputTable = formatDate(outputTable);
+
+  //creating a new sheet with name outputSheet and outputTable as its contents
+  createNewSheet(outputTable,outputSheet);
+  
+}
+
+//function to get the table data and set rowRange of JsonObj
+function getTable(tableData) {
+
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var objRowRange = {
+      'header' : 1,
+      'rowStart': 2,
+      'rowEnd': 2
+  }
+
+  //if range is not specified we consider entire data 
+  //else we use the range provided
+
+  if(tableData[0] === '') {
+    var range=sheet.getDataRange();
+    objRowRange.rowEnd = range.getLastRow();
+    objRowRange.header = objRowRange.rowEnd - range.getHeight() + 1;
+    objRowRange.rowStart = objRowRange.header + 1;
+  }
+  else {
+    var range = sheet.getRange(tableData[0]);
+    objRowRange.header = tableData[1];
+    objRowRange.rowEnd = range.getLastRow();
+    objRowRange.rowStart = objRowRange.rowEnd - range.getHeight()+1;
+    if(objRowRange.rowStart === objRowRange.header)
+      objRowRange.rowStart = objRowRange.rowStart+1;
+      
+    //assuming there is always one header row and atleast one data row 
+    //it may not be true always so error check required
+  }
+  
+  /*
+   * set rowRange
+   * JsonObj.rowRange is an object with keys header, rowStart and rowEnd
+   * header, rowStart and rowEnd are integers
+   */
+  JsonObj.rowRange = objRowRange;
+
+  //sending entire sheet data 
+  var rangeAll = sheet.getDataRange();
+  return rangeAll.getValues();
+}
+
+//function to call gcp function to obtain result from the received data
+function callGcpFunc() {
```

Comment:
Modify the function name , the name should imply what the gcp function is accomplishing.